generators/FairBoxGenerator.cxx

```diff
@@ -126,26 +114,38 @@ FairBoxGenerator& FairBoxGenerator::operator=(const FairBoxGenerator& rhs)
   fCosThetaIsSet = rhs.fCosThetaIsSet;
   fPtRangeIsSet = rhs.fPtRangeIsSet;
   fPRangeIsSet = rhs.fPRangeIsSet;
-  fPointVtxIsSet = rhs.fPointVtxIsSet;
-  fBoxVtxIsSet = rhs.fBoxVtxIsSet;
   fDebug = rhs.fDebug;
   fEkinRangeIsSet = rhs.fEkinRangeIsSet;
 
   return *this;
 }
 
+void FairBoxGenerator::SetXYZ(Double32_t x, Double32_t y, Double32_t z){
+    SetVertex(x,y,z, 0, 0, 0, kBox);
+}
+
+void FairBoxGenerator::SetBoxXYZ(Double32_t x1, Double32_t y1, Double32_t x2, Double32_t y2, Double32_t z){
+    Double_t X1  = TMath::Min(x1,x2);
+    Double_t X2 = TMath::Max(x1,x2);
+    Double_t Y1  = TMath::Min(y1,y2);
+    Double_t Y2 = TMath::Max(y1,y2);
+    Double_t dX = 0.5*(X2 - X1);
+    Double_t dY = 0.5*(Y2 - Y1);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'dY' is assigned a value that is never used.](https://app.codacy.com/manual/FairRootGroup/FairRoot/pullRequest?prid=5136019)